package/MDAnalysis/topology/GROParser.py

```diff
@@ -63,97 +63,4 @@
-        # resid, resname, name, index, (x,y,z)
-        with openany(self.filename) as inf:
-            next(inf)
-            n_atoms = int(next(inf))
-
-            # Allocate shizznizz
-            resids = np.zeros(n_atoms, dtype=np.int32)
-            resnames = np.zeros(n_atoms, dtype=object)
-            names = np.zeros(n_atoms, dtype=object)
-            indices = np.zeros(n_atoms, dtype=np.int32)
-
-            for i, line in enumerate(inf):
-                if i == n_atoms:
-                    break
-                try:
-                    resids[i] = int(line[:5])
-                    resnames[i] = line[5:10].strip()
-                    names[i] = line[10:15].strip()
-                    indices[i] = int(line[15:20])
-                except (ValueError, TypeError):
-                    raise IOError(
-                        "Couldn't read the following line of the .gro file:\n"
-                        "{0}".format(line))
-        # Check all lines had names
-        if not np.all(names):
-            missing = np.where(names == '')
-            raise IOError("Missing atom name on line: {0}"
-                          "".format(missing[0][0] + 3))  # 2 header, 1 based
-
-        # Fix wrapping of resids (if we ever saw a wrap)
-        if np.any(resids == 0):
-            # find places where resid hit zero again
-            wraps = np.where(resids == 0)[0]
-            # group these places together:
-            # find indices of first 0 in each block of zeroes
-            # 1) find large changes in index, (ie non sequential blocks)
-            diff = np.diff(wraps) != 1
-            # 2) make array of where 0-blocks start
-            starts = np.hstack([wraps[0], wraps[1:][diff]])
-
-            # remove 0 in starts, ie the first residue **can** be 0
-            if starts[0] == 0:
-                starts = starts[1:]
-
-            # for each resid after a wrap, add 100k (5 digit wrap)
-            for s in starts:
-                resids[s:] += 100000
-
-        # Guess types and masses
-        atomtypes = guessers.guess_types(names)
-        masses = guessers.guess_masses(atomtypes)
-
-        residx, (new_resids, new_resnames) = change_squash(
-                                (resids, resnames), (resids, resnames))
-
-        # new_resids is len(residues)
-        # so resindex 0 has resid new_resids[0]
-        attrs = [
-            Atomnames(names),
-            Atomids(indices),
-            Atomtypes(atomtypes, guessed=True),
-            Resids(new_resids),
-            Resnums(new_resids.copy()),
-            Resnames(new_resnames),
-            Masses(masses, guessed=True),
-            Segids(np.array(['SYSTEM'], dtype=object))
-        ]
-
-        top = Topology(n_atoms=n_atoms, n_res=len(new_resids), n_seg=1,
-                       attrs=attrs,
-                       atom_resindex=residx,
-                       residue_segindex=None)
-
-        return top
+from .c_GROParser import GROParser
```

Comment:
You need to build the C extension into the library via a `setup.py` file so that this doesn't cause issues during the build, which is also visible in CI.

For examples of MDA compiled extension building -- `git grep 'MDAExtension'`. I don't think we normally commit the Cython generated code to our repo. Probably best to just remove that file from version control & set up proper building of the extension.

Reply:
Oh, I did build it with setup.py but didn't know how to commit it right. Probably that might have caused the build error. The `.so` files are ignored anyway. So ones can you clarify what files to include after `setup.py`. I'm building with `python setup.py build_ext --inplace`.

Reply:
Ok, I get it. So I need to make an `MDAExtension` object for it in the `setup.py` file and cythonize list it there itself, right?

Reply:
Yup, CI is all green now, so I'll mark this resolved